include/libtorrent/disk_interface.hpp

```diff
@@ -170,8 +173,13 @@ namespace file_open_mode {
 			, char const* buf, std::shared_ptr<disk_observer> o
 			, std::function<void(storage_error const&)> handler
 			, disk_job_flags_t flags = {}) = 0;
-		virtual void async_hash(storage_index_t storage, piece_index_t piece, disk_job_flags_t flags
+		// if v2 is non-empty it must be at least large enough to hold all v2 blocks in the piece
+		virtual void async_hash(storage_index_t storage, piece_index_t piece, span<sha256_hash> v2
+			, disk_job_flags_t flags
 			, std::function<void(piece_index_t, sha1_hash const&, storage_error const&)> handler) = 0;
+		// async_hash2 computes the v2 hash of a single block
+		virtual void async_hash2(storage_index_t storage, piece_index_t piece, int offset, disk_job_flags_t flags
```

Comment:
is `offset` a byte offset or block offset? (I think the comment should say, and in the future it should be part of the type). I think all other offsets in the disk I/O code refers to byte offsets as just "offset"

Reply:
`offset` is bytes from the start of the piece, same as the other disk I/O functions.